app/models/assessment.rb

```diff
@@ -300,6 +300,36 @@ def dump_yaml
     File.open(asmt_yaml_path, "w") { |f| f.write(YAML.dump(sort_hash(serialize))) }
   end
 
+  # If the name field in a yaml for an assessment differs from the folder name
+  # then the name field for the assessment gets overridden, which then causes
+  # the yaml file to no longer be found, and then for cleanup to fail because
+  # the folder can no longer also be found. For example a folder with
+  # homework01 and a homework01.yml file but with a name field of homework02
+  # would cause assessment.name = homework02 so then it would try to find
+  # homework02/homework02.yml. Instead we rename the name field to match the folder
+  # name.
+  # Similarly for a bunch of other fields in the yaml, if they are not provided,
+  # then we may run into issues loading the assessment
+  def validate_yaml(yaml_map)
+    unless yaml_map["general"]
+      raise "General section missing in yaml"
```

Comment:
Is this caught by importAssessmentsFromFileSystem? It seems to catch StandardError and I think this is a TypeError.